ci/envoy-sync-receive.sh

```diff
@@ -0,0 +1,124 @@
+# than creating new ones that superceed the old one(s) each time.
+DST_NEW_BRANCH_NAME="auto-merge-$(echo "${SRC_BRANCH_NAME}" | tr /. -)"
+
+# Set the default remote for the gh command
+gh repo set-default "${DST_REPO_PATH}"
+
+# Perform the merge using --no-ff option to force creating a merge commit
+if git merge --no-ff --log=10000 --signoff -m "${TITLE}" \
+                "upstream/${SRC_BRANCH_NAME}" > "${SCRATCH}/mergeout"; then
+    DST_NEW_HEAD_SHA="$(git rev-parse HEAD)"
+    if [[ "${DST_NEW_HEAD_SHA}" != "${DST_HEAD_SHA}" ]]; then
+        git push --force origin "HEAD:${DST_NEW_BRANCH_NAME}"
+        PR_COUNT=$(gh pr list --head "${DST_NEW_BRANCH_NAME}" \
+                              --base "${DST_BRANCH_NAME}" \
+                              --state open | wc -l)
+        if [[ ${PR_COUNT} == 0 ]]; then
+            PR_URL=$(gh pr create --head "${DST_NEW_BRANCH_NAME}" \
+                                  --base "${DST_BRANCH_NAME}" \
+                                  --title "${TITLE}" \
+                                  --body "Generated by $(basename "$0")")
+            MERGE_OUTCOME="Created ${PR_URL}"
+        else
+            PR_ID=$(gh pr list --head "${DST_NEW_BRANCH_NAME}" \
+                               --base "${DST_BRANCH_NAME}" \
+                               --state open | head -1 | cut -f1)
+            PR_URL="https://github.com/${DST_REPO_PATH}/pull/${PR_ID}"
+            MERGE_OUTCOME="Updated ${PR_URL}"
+        fi
+    else
+        MERGE_OUTCOME="No changes"
+    fi
+    notice "${TITLE} successful (${MERGE_OUTCOME})"
+    # Close any related issues with a comment describing why
+    for ISSUE_ID in $(gh issue list -S "${TITLE} failed" | cut -f1); do
+        ISSUE_URL="https://github.com/${DST_REPO_PATH}/issues/${ISSUE_ID}"
+        gh issue close "${ISSUE_URL}" --comment "Successful ${TITLE} (${MERGE_OUTCOME})"
+        notice "Closed ${ISSUE_URL}"
+    done
+else # merge fail
+    notice "${TITLE} failed"
+    ISSUE_COUNT=$(gh issue list -S "${TITLE} failed" | wc -l)
+    if [[ ${ISSUE_COUNT} == 0 ]]; then
+        ISSUE_URL=$(gh issue create --title "${TITLE} failed" --body "${TITLE} failed")
+        ISSUE_ID="$(basename "${ISSUE_URL}")"
+        ISSUE_OUTCOME="Created"
+    else
+        ISSUE_ID="$(gh issue list -S "${TITLE} failed sort:created-asc" | tail -1 | cut -f1)"
+        ISSUE_URL="https://github.com/${DST_REPO_PATH}/issues/${ISSUE_ID}"
+        ISSUE_OUTCOME="Updated"
+    fi
+    COMMENT_URL=$(\
+        gh issue comment "${ISSUE_URL}" --body-file - <<-EOF
+			Failed to ${TITLE}
+			
+			Upstream   : [${SRC_HEAD_SHA}](https://github.com/${SRC_REPO_PATH}/commit/${SRC_HEAD_SHA})
+			Downstream : [${DST_HEAD_SHA}](https://github.com/${DST_REPO_PATH}/commit/${DST_HEAD_SHA})
+			
+			\`\`\`
+			$(cat "${SCRATCH}/mergeout" || true)
+			\`\`\`
+		EOF
+    )
+    notice "${ISSUE_OUTCOME} ISSUE#${ISSUE_ID} (${COMMENT_URL})"
+    exit 1
```

Comment:
nit: do we need to exit the job with an error? Strictly speaking the job succeeded. The merge failed probably because of a merge conflict, not because this script failed.

Reply:
I figured it was best to make merge failures is as visible as possible so that they don't get overlooked.
Also, the top level workflow executes with a `fail-fast` strategy of `false` so that a merge failure on one branch won't prevent successful merges from completing on other branches.